### ISSUE TITLE
Storybook: Button/ButtonGroup with icon needs a name/label

### ISSUE THREAD

**Issue by marcysutton** [forem/forem#11723]
**Describe the bug**

In Storybook, the Button component Icon variant is missing a label to expose an accessible name for the button element. This occurs on standalone buttons and Button Groups.

There are a couple of solutions for this, depending on how resilient the component needs to be:

- SVG Icons in all examples/tests/usage could include a `<title>` to expose an accessible name on the parent button, but this depends on authors remembering to do so. 
- The Button component could also make labeling easier by taking an `iconName` prop (or similar), falling back to the SVG title if it exists. I couldn't think of a good fallback if neither of those were present.

**To Reproduce**

1. Go to https://storybook.forem.com/?path=/story/components-buttongroup--text-icon
2. Click on the Accessibility tab
3. See the error report for missing text on the checkmark icon button. You can also inspect it in the DevTools to see there is no label on the button, and no SVG title.

**Expected behavior**

The Storybook Button and Button Group examples should be accessible. The Button component could take an iconName, and/or the inline SVG snippets could include `<title>` elements to act as accessible names for the parent buttons.

**Screenshots**

<img width="1392" alt="Storybook screenshot of Button Group component with failing Accessibility tab" src="https://user-images.githubusercontent.com/1045233/100972497-c6139e00-34ed-11eb-8760-5f16ed5f92cb.png">

**Desktop (please complete the following information):**

- OS, version:  All
- Browser, version: All

**Smartphone (please complete the following information):**

- Device: All
- OS, version: All
- Browser, version: All

**Additional context**

I'll file additional Component issues separately, as I'm going through Storybook as a whole.

**Comment by lisasy** [comment]
I'm tagging @nickytonline here for visibility. 👀

**Comment by nickytonline** [comment]
Thanks for raising @marcysutton. I'll take a peek at this.

**Comment by YuriDevAT** [comment]
If this issue is still open, I would like to work on it 🙂

**Comment by nickytonline** [comment]
I no longer work at Forem, but looking at the code in the devtools, and it looks like this is still an issue. I think adding a title to the SVG makes sense, but I'll let the current devs at Forem chime in.

![screenshot of the button icon in the button group not having an accessible name for itself](https://github.com/forem/forem/assets/833231/df3773e4-3309-4fc7-8400-4c1097dbc664)
